pep-9999.rst

```diff
@@ -0,0 +1,506 @@
+.. _Rust's Cargo: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html
+.. _comparison document: https://github.com/uranusjr/packaging-metadata-comparisons/blob/master/topics/dependency-entries.md
+
+
+Specification
+=============
+
+As in :pep:`621`, if metadata is improperly specified then tools MUST raise an
+error. The metadata MUST conform to the `toml`_ specification.
+
+To reduce confusion with this document being a specification for specifying
+dependencies, the word "requirement" is used to mean a :pep:`508` dependency
+specification.
+
+The following tables are added to the added to the ``project`` table specified
+in :pep:`621`.
+
+.. _toml: https://toml.io/
+
+dependencies
+------------
+
+Format: table
+
+The keys inside this table are the names of the required distribution. The
+values can have one of the following types:
+
+- string: the requirement is defined only by a version requirement, with same
+  specification as ``version`` in the requirement table, except allowing the
+  empty string ``""`` to place no restriction on the version.
+
+- table: a requirement table.
+
+- array: an array of requirement tables. It is an error to specify an empty
+  array ``[]`` as a value.
+
+.. _requirement-spec:
+
+Requirement table
+^^^^^^^^^^^^^^^^^
+
+The keys of the requirement table are as follows (all are optional):
+
+- ``version`` (string): a :pep:`440` version specifier, which is a comma-
+  delimited list of version specifier clauses. The string MUST be non-empty.
+
+- ``extras`` (array of strings): a list of :pep:`508` extras declarations for
+  the distribution. The list MUST be non-empty.
+
+- ``markers`` (string): a :pep:`508` environment marker expression. The string
+  MUST be non-empty.
+
+- ``url`` (string): the URL of the artifact to install and satisfy the
+  requirement. Note that ``file://`` is the prefix used for packages to be
+  retrieved from the local filesystem.
+
+- ``git``, ``hg``, ``bzr`` or ``svn`` (string): the URL of a VCS repository
+  (as specified in :pep:`440`)
+  to clone, whose tree will be installed to satisfy the requirement. Further
+  VCS keys will be added via amendments to :pep:`610`, however tools can opt to
+  support other VCS's using their command-line command prior to the acceptance
+  of the amendment.
+
+- ``revision`` (string): the identifier for a specific revision of the
+  specified VCS repository to check-out before installtion. Users MUST only
+  provide this when one of ``git``, ``hg``, ``bzr``, ``svn``, or another VCS
+  key is used to identify the distribution to install. Revision identifiers are
+  suggested in :pep:`610`.
+
+At most one of the following keys can be specified simultaneously, as they
+logically conflict with each other in the requirement: ``version``, ``url``,
+``git``, ``hg``, ``bzr``, ``svn``, and any other VCS key.
+
+An empty requirement table ``{}`` places no restriction on the requirement.
```

Comment:
Is this  meant to be equivalent to the empty string? So there's two ways to list a dependency with no restrictions?

Reply:
Yes. The empty string (or the string `"*"`) was fairly heavily pushed for as placing no constraint, while there was no comment on the empty table. What's worse is that the empty array is not allowed. This is something to bring up in the discussion.